doc/classes/Dictionary.xml

```diff
@@ -214,6 +214,24 @@
 				Returns the corresponding value for the given [param key] in the dictionary. If the [param key] does not exist, returns [param default], or [code]null[/code] if the parameter is omitted.
 			</description>
 		</method>
+		<method name="get_key_at_index" qualifiers="const">
+			<return type="Variant" />
+			<param index="0" name="index" type="int" />
+			<description>
+				Returns the key for the [param index]-th element (by order of insertion) in the dictionary. Negative indices are supported (i.e., counting from the back). If the index is out of bounds, returns [code]null[/code].
+				[codeblock]
+				var dict = {"A": 0, "B": 1, "C": 2, "D": 3}
+
+				print(dict.get_key_at_index(0))  # Prints "A"
+				print(dict.get_key_at_index(1))  # Prints "B"
+
+				dict["E"] = 4
+				print(dict.get_key_at_index(4))  # Prints "E"
+				print(dict.get_key_at_index(-1)) # Prints "E"
+				[/codeblock]
+				[b]Note:[/b] This method runs in O(1) time for elements near the beginning or end of the dictionary, but is considerably slower for retrieving elements in the middle of large dictionaries. To retrieve all keys in order of insertion as an array, use [method keys].
```

Comment:
You can't really approximate `O(n)` as `O(1)` if n is low enough. Or at least I don't think that's how it works :)

Perhaps you could phrase it somewhat like "This method iterates the dictionary to find the element at the given index. This is fast for elements near the start or end of the dictionary, but slower for elements in the middle."

Reply:
I meant to refer to a constant index. But I like your version much better. Thanks!